### PR TITLE
Add repr() calls to "invalid value" exception message in StrConverter.to_bool

### PR DESCRIPTION
**Pull request by ptmcg**
After upgrading to 4.0, we are getting the following exceptions in our build pipeline:

    File "/tmp/tmpw_2jv6x3/python-test-env/lib/python3.9/site-packages/tox/config/cli/parser.py", line 293, in add_color_flags
        if converter.to_bool(os.environ.get("NO_COLOR", "")):
    File "/tmp/tmpw_2jv6x3/python-test-env/lib/python3.9/site-packages/tox/config/loader/str_convert.py", line 90, in to_bool
        raise TypeError(f"value {value} cannot be transformed to bool, valid: {', '.join(StrConvert.VALID_BOOL)}")
    TypeError: value  cannot be transformed to bool, valid: , 0, 1, false, no, off, on, true, yes

We cannot reproduce the issue locally, else we would have more debugging tools and options.

This PR adds !r to display the repr() of the bad value, as well as repr()'s the values in VALID_BOOL, to help diagnose issues when running locally is not an option.

# Thanks for contribution

Please, make sure you address all the checklists (for details on how see
[development documentation](http://tox.readthedocs.org/en/latest/development.html#development))!

- [x] ran the linter to address style issues (`tox -e fix_lint`)
- [X] wrote descriptive pull request text
- [x] ensured there are test(s) validating the fix
- [X] added news fragment in `docs/changelog` folder
- [ ] updated/extended the documentation

### CODE REVIEW

File: src/tox/config/loader/str_convert.py [hunk 1]
```diff
@@ -81,13 +81,16 @@ def to_env_list(value: str) -> EnvList:
...
+        norm = str(value).strip().lower()
         if norm in StrConvert.TRUTHFUL_VALUES:
             return True
         elif norm in StrConvert.FALSE_VALUES:
             return False
         else:
-            raise TypeError(f"value {value} cannot be transformed to bool, valid: {', '.join(StrConvert.VALID_BOOL)}")
+            raise TypeError(
+                f"value {value!r} cannot be transformed to bool,"
+                f" valid values: {', '.join(repr(vb) for vb in StrConvert.VALID_BOOL)}",
```
Comment (gaborbernat) [inline src/tox/config/loader/str_convert.py:92]: I don't want to use repr for valid values

**Review by gaborbernat — CHANGES_REQUESTED** [review]

File: src/tox/config/loader/str_convert.py (hunk 1 above)
Comment (ptmcg) [reply src/tox/config/loader/str_convert.py:92]: When one of the valid values is the empty string, it makes for a weird message.

**Comment by gaborbernat** [comment]
Also needs a test. My guess is you're passing an empty string for boolean.

**Comment by ptmcg** [comment]
Thanks for the quick review, by the way!

That was our thought as well (passing in an empty string), but the logic in the code should be able to handle an empty string, as that appears to be one of the values in VALID_BOOL.

I looked for other tests, but did not see any for this method.

**Comment by gaborbernat** [comment]
> I looked for other tests, but did not see any for this method.

https://github.com/tox-dev/tox/blob/main/tests/config/test_sets.py#L92

**Comment by ptmcg** [comment]
Ah, I was looking for something specifically testing to_bool(). Found the appropriate test, updated the expected error message.

**Pull request merged**